### PR TITLE
Edit stochastic docs

### PR DESCRIPTION
**Pull request by Leo-am**
**Description**

Edited the user guide documentation for Stochastic solvers. Text, equations, and example updated.

**Related issues or PRs**
Related to issue #1706.

**Changelog**

- Improved explanation of stochastic equation solvers.
- Added links for notebooks with examples for stochastic solvers.
- Added links to API documentation for smesolve, ssesolve, general_stochastic.
- New example highlighting the differences between result.expect and result.measurement.
- Added reference to Quantum Measurement and Control, Wiseman & Milburn (2009).

### CODE REVIEW

**Comment by Ericgig** [comment]
Thank you Leo-am, I took a quick look and it's nice.
We will do a proper review in January.

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 1]
```diff
@@ -6,109 +6,161 @@ Stochastic Solver
...
+where
 
 .. math::
-	:label: liouvillian
+    :label: dissipator
 
-	L(\rho(t)) = - i[H(t),\rho(t)]
-	             + \sum_n D(S_n, \rho)
-				 + \sum_i D(C_i, \rho),
+    D[A] \rho = \frac{1}{2} \left[2 A \rho C^\dagger
```
Comment (Ericgig) [inline doc/guide/dynamics/dynamics-stochastic.rst:80]: The `C^\dagger` should be `A^\dagger`.

**Review by Ericgig — CHANGES_REQUESTED** [review]
Thank you for the rewriting, look great.
The example is a lot better than what it was.
The equations are clear, but there are a few typos to fix.

File: doc/guide/dynamics/dynamics-stochastic.rst (hunk 1 above)
Comment (Leo-am) [reply doc/guide/dynamics/dynamics-stochastic.rst]: Typo fixed.

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 2]
```diff
@@ -6,109 +6,161 @@ Stochastic Solver
...
-    times = np.linspace(0.0, 10.0, 201)
-    psi0 = tensor(fock(2, 0), fock(10, 5))
-    a  = tensor(qeye(2), destroy(10))
-    sm = tensor(destroy(2), qeye(10))
+.. math::
+    :label: d2_def
 
-    H = 2*np.pi*a.dag()*a + 2*np.pi*sm.dag()*sm + 2*np.pi*0.25*(sm*a.dag() + sm.dag()*a)
-    data = ssesolve(H, psi0, times, sc_ops=[np.sqrt(0.1) * a], e_ops=[a.dag()*a, sm.dag()*sm], method="homodyne")
+    d_2 = S_n - \frac{e_n}{2}.
```
[the remark this answers is not in the document]
Comment (Leo-am) [reply doc/guide/dynamics/dynamics-stochastic.rst]: Change made.

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 3]
```diff
@@ -6,109 +6,161 @@ Stochastic Solver
...
+   e_n = \left<\psi(t)|S_n + S_n^\dagger|\psi(t)\right>
+
+In QuTiP, this equation can be solved using the function :func:`qutip.stochastic.ssesolve`, which is implemented by defining :math:`d_1` and :math:`d_{2,n}` from Equation :eq:`general_form` as
 
-Here :math:`\delta \omega` is a Wiener increment.
+.. math::
+    :label: d1_def
 
-In QuTiP, this is available with the function :func:`ssesolve`.
+    d_1 = -iH - \frac{1}{2} \left(S_n^\dagger S_n - e_n S_n + \frac{e_i^2}{4}  \right),
```
[the remark this answers is not in the document]
Comment (Leo-am) [reply doc/guide/dynamics/dynamics-stochastic.rst]: Change made.

**Comment by Leo-am** [comment]
Thanks for reviewing, Eric. All typos are fixed.

**Review by Ericgig — APPROVED** [review]

**Comment by hodgestar** [comment]
Thanks @Leo-am for making these docs much better and thank you @Ericgig for reviewing. Merging.

**Pull request merged**